src/main/java/org/mskcc/cmo/publisher/pipeline/config/BatchConfiguration.java

```diff
@@ -64,9 +68,10 @@
     @Autowired
     private Gateway messagingGateway;
 
-    @Autowired
-    public void initGatewayConnection() throws Exception {
+    @Bean
+    public Gateway messagingGateway() throws Exception {
```

Comment:
note: gateway is autowired above but it still needs to be provided to the job application context as a bean which is what's happening here